pkg/segment/query/processor/wherecommand.go

```diff
@@ -42,8 +43,16 @@ func (p *whereProcessor) Process(iqr *iqr.IQR) (*iqr.IQR, error) {
 	for _, field := range requiredFields {
 		values, err := iqr.ReadColumn(field)
 		if err != nil {
-			log.Errorf("where.Process: cannot get field values; field=%s; err=%v", field, err)
-			return nil, err
+			return nil, utils.TeeErrorf("where.Process: cannot get field values; field: %s; err: %v", field, err)
+		}
+		if values == nil {
+			// The field is not present, discard all the rows
+			// TODO: Fix evaluation for non existing fields
+			err := iqr.Discard(iqr.NumberOfRecords())
+			if err != nil {
+				return nil, utils.TeeErrorf("where.Process: Error while discarding rows due to col %v not found, err: %v", field, err)
```

Comment:
nit: you could also just return `nil, nil`, or make a new empty IQR and return that, but this also works